do not depend on Oscar

ThomasBreuer opened the pull request:
This is a version which can be used either with Oscar.jl or with Gapjm.jl.
The idea is to use Oscar.jl's polynomials if this package is already loaded
before GenericDecMats.jl, and to use Gapjm.jl's multivariate polynomials
if this package is already loaded.

(For technical reasons, Requires.jl is used.
However, we cannot use its feature that the functionality depending on
one of the two packages becomes available as soon as Oscar.jl or Gapjm.jl
gets loaded later on.
Note that the type `GenericDecompositionMatrix` can support only one
of the two types of polynomials.)

ThomasBreuer committed:
experimental: do not depend on Oscar

This is a version which can be used either with Oscar.jl or with Gapjm.jl.
The idea is to use Oscar.jl's polynomials if this package is already loaded
before GenericDecMats.jl, and to use Gapjm.jl's multivariate polynomials
if this package is already loaded.

(For technical reasons, Requires.jl is used.
However, we cannot use its feature that the functionality depending on
one of the two packages becomes available as soon as Oscar.jl or Gapjm.jl
gets loaded later on.
Note that the type `GenericDecompositionMatrix` can support only one
of the two types of polynomials.)

thofma commented on src/GenericDecMats.jl:
```diff
@@ -216,4 +90,21 @@ function generic_decomposition_matrices_overview()
 end
 
+function __init__()
+    if isdefined(Main, :Oscar)
```
```suggestion
```
Is this necessary? When I use Requires I don't have to do this.

ThomasBreuer replied:
The point is that I do not want to include both variants, because then it is not clear which polynomials will be used.
For example, if Gapjm.jl is loaded in advance then the corresponding functionality from GenericDecMats.jl gets loaded. But if `Requires.@require` has been called also for Oscar.jl, and Oscar gets loaded later on, then also the Oscar related file would get included.

thofma replied:
OK, I see. Thanks for the explanation.

thofma commented:
Why the duplication? Can't we use the same type if we remove the restriction with the package-dependent field type? So instead of ` vars::Vector{Gapjm.Mvp}` just `vars`. Then use Requires to add a constructor depending on the package loaded.

ThomasBreuer committed:
next iteration (which can be merged)

- avoid code duplication
- improved type of the `decmat` field in the Gapjm situation

ThomasBreuer commented:
@thofma Thanks for the hint. I have updated the pull request.
